### PR TITLE
#170947562 Users get notifications for approvals

### PR DESCRIPTION
**Pull request by Baraka-Mugisha**
#### What does this PR do?
add a `Users get notifications for approved requests` feature

#### Description of Task to be completed?
- added a table for notification
- installed socket.io
- test the feature
- documented the feature
- internationalized all messages

#### How should this be manually tested?
- clone this repo
- run npm install to install all dependencies
- run the server (npm run reset:dev)
- make a put request to `/api/v1/trips/{requestId}/confirm`
- you will get an email with a link to the approved request

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#170947562](https://www.pivotaltracker.com/story/show/170947562)

#### Screenshots (if appropriate)
Screen Shot 2020-03-11 at 12 06 12

#### Questions:
- N/A

### CODE REVIEW

**Review by Ntare22 — CHANGES_REQUESTED** [review]
Test coverage has drastically gone down... could you please write more tests especially for io.js file? Make sure not to hit an external API?

File: src/controllers/tripsController.js [hunk 1]
```diff
@@ -255,6 +259,17 @@ export default class requestController {
...
       }
       const updatedRequest = await request.update({ confirm: true });
+
+      const requester = await db.User.findOne({
+        where: {
+          email: updatedRequest.email,
+        },
+        attributes: ['id', 'email', 'password', 'firstName', 'isVerified', 'role'],
+      });
+
```
Comment (bbaime98) [inline src/controllers/tripsController.js]: why are you rewriting this query? I think we do have it already.

**Comment by Baraka-Mugisha** [comment]
> Test coverage has drastically gone down... could you please write more tests especially for io.js file? Make sure not to hit an external API?

Yeah, I made sure of that. the test coverage has now also come back to normal

**Review by izzett222 — APPROVED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

File: public/index.html [hunk 2]
```diff
@@ -25,13 +33,52 @@
...
+            for(obj in parsedData.notif) { 
+                console.log(parsedData.notif);
+                let node = document.createElement('div');
+                node.innerHTML = `<a href = ${parsedData.notif[obj].link} target="_blank"><li class = "notifs">${parsedData.notif[obj].content}</li></a>`
+                notifDiv.appendChild(node)
+        }});
+
         socket.on('notification', (data) => {
-            socket.emit('messageFromClient', 'This is from the client')
+            console.log(data);
```
Comment (izzett222) [inline public/index.html]: @Baraka-Mugisha can you remove console in this file
```suggestion
```

File: src/controllers/tripsController.js (hunk 1 above)
Comment (Baraka-Mugisha) [reply src/controllers/tripsController.js]: I removed it.

Comment (Baraka-Mugisha) [reply src/controllers/tripsController.js]: I didn't recognize that we added user_id attribute to the Requests table before

**Review by izzett222 — CHANGES_REQUESTED** [review]

**Review by izzett222 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

**Review by bbaime98 — DISMISSED** [review]

**Review by izzett222 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]
LGTM

**Review by izzett222 — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

**Review by Ntare22 — APPROVED** [review]

**Pull request merged**